src/main/java/software/amazon/smithy/gradle/tasks/SmithyTagsTask.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.gradle.tasks;
+
+import java.util.Set;
+import java.util.TreeSet;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.java.archives.Attributes;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.internal.logging.text.StyledTextOutput;
+import org.gradle.internal.logging.text.StyledTextOutputFactory;
+import org.gradle.jvm.tasks.Jar;
+import software.amazon.smithy.gradle.SmithyExtension;
+import software.amazon.smithy.gradle.SmithyUtils;
+
+/**
+ * Adds Smithy-Tags to the JAR built by the {@link Jar} Task.
+ * This Task does not define any outputs. So it will always be executed before the Jar Task.
+ */
+public class SmithyTagsTask extends DefaultTask {
+
+    private Set<String> tags = new TreeSet<>();
+
+    /**
+     * Get the tags that are added to the JAR.
+     *
+     * <p>These tags are placed in the META-INF/MANIFEST.MF attribute named
+     * "Smithy-Tags" as a comma separated list. JARs with Smithy-Tags can be
+     * queried when building projections so that the Smithy models found in
+     * each matching JAR are placed into the projection JAR.
+     *
+     * @return Returns the Smithy-Tags values that will be added to the created JAR.
+     */
+    @Input
```

Comment:
I am unsure if this is needed. Seems to work without it in my testing, but maybe there's a usecase where keeping this is still good. So left it there.

Reply:
It's just for use when programmatically interacting with the task. Technically this is a breaking change since these are being moved to a different task, but since the gradle plugin isn't GA yet it's fine. We'll just need to make sure we do a minor bump next time we release.